### ISSUE TITLE
AttributeError: module 'torchsparse_cuda' has no attribute 'hash_forward'

### ISSUE THREAD

**Issue by sandeepnmenon** [mit-han-lab/e3d#23]
I am trying to run the model inference code using the pretrained model using visualise.py. However at the line below
https://github.com/mit-han-lab/e3d/blob/db65d6c968a12d30a4caa2715ef6766ec04d7505/spvnas/visualize.py#L189
I am getting the following error
```
outputs = model(inputs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/docker/e3d/spvnas/core/models/semantic_kitti/spvcnn.py", line 194, in forward
    x0 = initial_voxelize(z, self.pres, self.vres)
  File "/opt/docker/e3d/spvnas/core/models/utils.py", line 19, in initial_voxelize
    pc_hash = spf.sphash(torch.floor(new_float_coord).int())
  File "/usr/local/lib/python3.6/dist-packages/torchsparse/nn/functional/hash.py", line 44, in sphash
    return hash_gpu(idx)
  File "/usr/local/lib/python3.6/dist-packages/torchsparse/nn/functional/hash.py", line 12, in forward
    return torchsparse_cuda.hash_forward(idx.contiguous())
AttributeError: module 'torchsparse_cuda' has no attribute 'hash_forward'
```

Environment details
Docker image: nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04
Installed all prerequisites as mentioned in the repo
Torch 1.7.0
torchsparse 1.1.0
Cuda 10.2
GPU GeForce GTX TITanX 12GB

**Comment by YoushaaMurhij** [comment]
pip install --upgrade git+https://github.com/mit-han-lab/torchsparse.git

**Comment by sandeepnmenon** [comment]
> pip install --upgrade git+https://github.com/mit-han-lab/torchsparse.git

I installed using this command only in my Dockerfile. I even tried running the same command twice in the dockerfile. Still the same issue.

**Comment by YoushaaMurhij** [comment]
Try it in terminal after running. Not in dockerfile. I faced the smae problem!
for dockerfile I don't know how to fix it. I think it is a cuda related issue.

**Comment by sandeepnmenon** [comment]
Found the issue!
Yes it is working in terminal for me and even after going in the interactive mode of the docker image and reinstalling there again.
Docker build does not use GPU context by default while building docker images. Hence, the install of torchsparse is not happening in GPU context. That is why the cuda issue is coming.
I solved it by forcing docker to use gpu and cuda while building by setting the following variables before running the torchsparse instlal

```
ENV FORCE_CUDA="1"
ARG TORCH_CUDA_ARCH_LIST="Kepler;Kepler+Tesla;Maxwell"
ENV TORCH_CUDA_ARCH_LIST="${TORCH_CUDA_ARCH_LIST}"
RUN python3 -m pip install --upgrade git+https://github.com/mit-han-lab/torchsparse.git
```

Apologies for taking your time. Hope this issue serves as a reference for others who face similar issue.
